shell/packages/sandstorm-ui-powerbox/powerbox-client.js

```diff
@@ -193,133 +133,103 @@ SandstormPowerboxRequest = class SandstormPowerboxRequest {
+      cardData.iconSrc = pkg ? this._db.iconSrcForPackage(pkg, "grain") : "";
+      cardData.lastUsed = grain.lastUsed;
 
-    return cards;
-  }
+      // Because Blaze always invokes functions when referenced as values from the data context, we
+      // need to double-wrap this callback.
+      cardData.callback = () => () => {
+        return this.selectGrain(cardData, grain.cachedViewInfo || {});
+      };
 
-  completeNewIpNetwork() {
-    Meteor.call("newFrontendRef",
-      this._requestInfo.sessionId,
-      { ipNetwork: true },
-      (err, token) => {
-        if (err) {
-          console.log(err);
-          this._error.set(err.toString());
-        } else {
-          this._completed = true;
-          this._requestInfo.source.postMessage({
-            rpcId: this._requestInfo.rpcId,
-            token: token,
-            descriptor: {
-              tags: [
-                { id: "12214421258504904768" }, // IpNetwork
-              ],
-              quality: "acceptable",
-            },
-          }, this._requestInfo.origin);
-          // Completion event closes popup.
-          this._requestInfo.onCompleted();
-        }
-      }
-    );
-  }
+      return;
+    }
 
-  ipInterfaceCardData() {
-    const cards = [];
-    if (this._db.isAdmin()) {
-      cards.push({
-        _id: "frontendref-ipinterface",
-        title: "Admin: grant all incoming network access",
-        iconSrc: "/settings.svg",
-        callback: () => {
-          // Because Blaze always invokes functions when referenced as values from the data context, we
-          // need to double-wrap this callback.
-          return () => {
-            this.completeNewIpInterface();
-          };
-        },
-      });
+    // Look for an ApiToken.
+    const apiToken = this._db.collections.apiTokens.findOne(
+        { grainId: cardData.grainId, "owner.user": { $exists: true } });
+    if (apiToken) {
+      const ownerData = apiToken.owner.user;
+      const grainInfo = ownerData.denormalizedGrainMetadata;
+      const staticAssetHost = this._db.makeWildcardHost("static");
+
+      cardData.title = ownerData.title;
+      cardData.appTitle =
+          (grainInfo && grainInfo.appTitle && grainInfo.appTitle.defaultText) || "";
+      cardData.iconSrc = (grainInfo && grainInfo.icon && grainInfo.icon.assetId) ?
+          (window.location.protocol + "//" + staticAssetHost + "/" + grainInfo.icon.assetId) :
+          Identicon.identiconForApp(
+              (grainInfo && grainInfo.appId) || "00000000000000000000000000000000");
+      cardData.lastUsed = ownerData.lastUsed;
+      cardData.apiTokenId = apiToken._id;
+
+      cardData.callback = () => () => {
+        Meteor.call("getViewInfoForApiToken", grainCard.tokenId, (err, result) => {
+          if (err) {
+            console.log(err);
+            this._error.set(err.toString());
+          } else {
+            this.selectGrain(grainCard, result || {});
```

Comment:
Not totally sure. I think it's usually true, but let's say you create a grain that fails before returning ViewInfo. Presumably you'll be left with a grain table entry with no cachedViewInfo. Better to be safe here.